### PR TITLE
Improvements to mesh tools

### PR DESCRIPTION
**Pull request by softmattertheory**
* MeshPruner added that enables coarsening of meshes. 

* Improvements to MeshMerge, which now doesn't elements with duplicate vertices. 

* Improvements to selection refinement, which now adds in additional vertices generated in the process of refinement. 

* Refinement of 3D elements. 

* Improvements to documentation

### CODE REVIEW

**Review by joshichaitanya3 — APPROVED** [review]
Looks great! I have highlighted some typos.

**Comment by joshichaitanya3** [comment]
Does the MeshMerge improvement fix #149 ?

**Comment by joshichaitanya3** [comment]
Oh, we also need a test for mesh pruning.

File: morpho5/docs/builtin.md [hunk 1]
```diff
@@ -8,6 +8,37 @@ Morpho provides a number of built-in functions.
...
+    import constants 
+    print sin(Pi/2) // expect: 1
+
+## sinh
+[tagsinh]: # (sinh)
+
+Returns the hyperbolic sine of an angle in radians.
+
+    import constants 
+    print sin(0) // expect: 0
```
[the remark this answers is not in the document]
Comment (softmattertheory) [reply morpho5/docs/builtin.md:25]: Fixed!

File: morpho5/docs/meshtools.md [hunk 2]
```diff
@@ -124,3 +172,58 @@ To fix this add the vertices first.
...
+
+    var mp = MeshRefiner(mesh)
+
+or with a list of objects that can include a `Mesh` as well as `Field`s and `Selection`s.
+
+    var mp = MeshRefiner([mesh, field, selection ... ])
+
+To perform the refinement, call the `prune` method with a `Selection`,
+
+    var dict = mp.refine(select)
```
[the remark this answers is not in the document]
Comment (softmattertheory) [reply morpho5/docs/meshtools.md:211]: It's actually NOT optional!

**Comment by softmattertheory** [comment]
> Does the MeshMerge improvement fix #149 ?

Alas, not quite yet. I will resolve this shortly.

**Pull request merged**